addons/repair/models/stock_move.py

```diff
@@ -175,8 +175,9 @@ def _set_repair_locations(self):
                 m.location_id, m.location_dest_id = m._get_repair_locations(line_type)
 
     def _should_be_assigned(self):
-        res = super(StockMove, self)._should_be_assigned()
-        return bool(res and not (self.repair_id and self.repair_line_type))
+        if self.repair_id:
```

Comment:
Given the previous check, shouldn't it be this instead?
```suggestion
        if self.repair_id and self.repair_line_type:
```

Reply:
no, because repair_line_type is set only for parts-related moves
the repaired product has no line type
anyway, both moves should not be taken in a new picking as the repair order is considered a one (just like manufacturing orders)
this is because the previous code considered the line type that the repaired product's move was missed...

Reply:
Indeed you're right, that would completely defeat the point. I misunderstood what was the issue.

Reply:
test added